pkg/cmd/autocomplete.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2024 Daytona Platforms Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package cmd
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"github.com/spf13/cobra"
+)
+
+var CompletionCmd = &cobra.Command{
+	Use:     "completion [bash|zsh|fish|powershell]",
+	Short:   "Adds completion script for your shell enviornment",
+	Args:    cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		shell := args[0]
+		homeDir, err := os.UserHomeDir()
+		if err != nil {
+			fmt.Printf("Error finding user home directory: %s\n", err)
+			return
+		}
+
+		var filePath, profilePath string
+		switch shell {
+		case "bash":
+			filePath = filepath.Join(homeDir, ".daytona.completion_script.bash")
+			profilePath = filepath.Join(homeDir, ".bashrc")
+		case "zsh":
+			filePath = filepath.Join(homeDir, ".daytona.completion_script.zsh")
+			profilePath = filepath.Join(homeDir, ".zshrc")
+		case "fish":
+			filePath = filepath.Join(homeDir, ".config", "fish", "daytona.completion_script.fish")
+			profilePath = filepath.Join(homeDir, ".config", "fish", "config.fish")
+		case "powershell":
+			filePath = filepath.Join(homeDir, "daytona.completion_script.ps1")
+			profilePath = filepath.Join(homeDir, "Documents", "WindowsPowerShell", "Microsoft.PowerShell_profile.ps1")
+		default:
+			fmt.Println("Unsupported shell type. Please use bash, zsh, fish, or powershell.")
+			return
+		}
+
+		file, err := os.Create(filePath)
+		if err != nil {
+			fmt.Printf("Error creating completion script file: %s\n", err)
+			return
+		}
+		defer file.Close()
+
+		switch shell {
+		case "bash":
+			err = cmd.Root().GenBashCompletion(file)
+		case "zsh":
+			err = cmd.Root().GenZshCompletion(file)
+		case "fish":
+			err = cmd.Root().GenFishCompletion(file, true)
+		case "powershell":
+			err = cmd.Root().GenPowerShellCompletionWithDesc(file)
+		}
+
+		if err != nil {
+			fmt.Printf("Error generating completion script: %s\n", err)
+			return
+		}
+
+		sourceCommand := fmt.Sprintf("source %s\n", filePath)
```

Comment:
<img width="442" alt="Screenshot 2024-03-09 at 12 36 51" src="https://github.com/daytonaio/daytona/assets/26512078/0ab6f56b-79a9-4fec-9b12-8086a2ae664d">

Please add a new line at the start of this command.

If the profile file does not end in a new line, then your code will produce something like the screenshot above.